tests/console/yast2_samba.pm

```diff
@@ -230,9 +227,39 @@ sub run {
     send_key 'alt-o';
 
     wait_serial('yast2-samba-server-status-0', 60) || die "'yast2 samba-server' didn't finish";
+}
+
+sub gui_new {
+    assert_screen 'yast2_samba-workgroup';
+    send_key 'alt-n';
+    if (check_screen 'yast2_still_susefirewall2', 10) {
+        record_soft_failure 'bsc#1064405';
+        send_key 'alt-c';
+    }
+    wait_still_screen 3;
+    # from here we have new workflow because of changes in sles 15
+}
+
+
+sub run {
+    setup_ldap;
+    zypper_call("in samba yast2-samba-server");
+    script_run("yast2 samba-server; echo yast2-samba-server-status-\$? > /dev/$serialdev", 0);
+    # check Samba-Server Configuration got started
+    if (is_leap && !leap_version_at_least('15.0') || (is_sle && !sle_version_at_least('15'))) {
```

Comment:
I strongly recommend to surround both leap-checks with parentheses, i.e. write `if ((is_leap && !leap_version_at_least('15.0')) || (is_sle && !sle_version_at_least('15'))) {`